### DISCUSSION TITLE
IMediaFile doesn't include the FullPath

### DISCUSSION THREAD

**Discussion by raulandresduque** [dimonovdd/Xamarin.MediaGallery#37] [Q&A] [answered]
### Description
It is not possible to get the path where the file(s) where selected

### Actual Behavior
IMediaFile doesn't include the FullPath

### Expected behavior
Have the FullPath or at least the Path, it makes the plugin unusable sine how are you going to access the file later, you will need the FullPath

### Steps to reproduce the behavior
1. Select a file (PickAsync)
2. Inspect IMediaFile collection returned
3. IMediaFile doesn't contain the path of the file

### Screenshots or Videos
N/A

### Reproduction Link
N/A

### Configuration
 - Device: Any
 - OS Version: Any
 - Plugin Version: 1.0.0

**Comment by dimonovdd** [comment, marked as answer]
@raulandresduque  Hi
It is not possible. Android and iOS have security policies that protect user data, so we can't get the real path to the file.
You can also view the documentation update in Xamarin.Essentials:
- https://github.com/xamarin/Essentials/pull/1759
- https://github.com/MicrosoftDocs/xamarin-docs/pull/3290

**Comment by hplos** [comment]
This can be done if the plugin returns the AssetIdentifier (iOS) and the uri (android). No permissions needed. Don't know exactly about android, but for iOS, if you want to do anything this that asset, you need permissions. Thats not included in this plugin.

Code to retrieve asset in iOS

`var assets = PHAsset.FetchAssetsUsingLocalIdentifiers(new string[] { mediaIdentifier }, new PHFetchOptions());`
`if (assets.firstObject is PHAsset asset)`
`{`
    `PHPhotoLibrary.SharedPhotoLibrary.PerformChanges(() =>`
`    {`
` //in my case I want to remove the media`
`        PHAssetChangeRequest.DeleteAssets(new PHAsset[] { asset });`
`    }, (success, error) => { });`
`}`

**Reply by dimonovdd** [reply]
This is a cross-platform plugin that has certain limitations due to the difference in platforms. But if you know how to solve this problem, everyone would be happy to have your pull request. Also, your sample code does not return the path to the real file anywhere, which can be processed through `System.IO`
Also pay attention to this:
https://github.com/xamarin/Essentials/issues/1746

**Reply by hplos** [reply]
I'm sorry for my bad example (in my app I don't need the full path, I just want to delete the asset)

`public string GetFullPath(string assetIdentifier)`
`{`
`    var result = string.Empty;`
`    var assets = PHAsset.FetchAssetsUsingLocalIdentifiers(new string[] { mediaIdentifier }, new PHFetchOptions());`
`    if (assets.firstObject is PHAsset asset)`
`    {`
`        PHImageManager imageManager = new PHImageManager();`
`        var options = new PHVideoRequestOptions();`

`        imageManager.RequestAvAsset(asset, options, `
`        (avAsset, audiomix, info) =>`
`        {`
`            if (avAsset != null)`
`            {`
`                result = ((AVUrlAsset)avAsset).Url.ToString();`
`            }`
`        });`
`    }`
`   `
`    return result;`
`}`

At this point I don't have a solution for Android. I will spend some time to check this on android when I can.

I know that you need permissions to get the full path or performe changes to the asset. With the solution in my pull request, the developer can choose to do so or not. I think that with this solution, the developer has full access if wanted. If not, nothing changes.

**Comment by raulandresduque** [comment]
Thank you @hplos.

Yes, other addons like xamarin essentials or James Montenegro return the URL so, beyond the security restrictions, it is possible to get with the correct rights granted to the APP.

**Reply by dimonovdd** [reply]
You can view the links I attached, the specified path will not be the actual path to the file.
